src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Utilities/NullableHelpers/NullableHelpers.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static class NullableHelpers
+    {
+        /// <summary>
+        /// Gets the declared symbol and root operation from the passed in declarationSyntax and calls
+        /// <see cref="IsSymbolAssignedPossiblyNullValue(SemanticModel, IOperation, ISymbol)"/>. Note
+        /// that this is bool and not bool? because we know that the symbol is at the very least declared, 
+        /// so there's no need to return a null value. 
```

Comment:
I'm a bit confused by this; what does something just like `string? s` give in this case?

Reply:
Do you mean just the declaration `string? s;`?

Reply:
Yep. I guess it wasn't entirely clear to me why you wouldn't do the nullable return like you did for the other function in that case. Or at least, the comment here might need to be a bit clearer?

Reply:
In IsSymbolAssignedPossiblyNullValue the only reason to return null is if there are no references at all to the given symbol. Since we're getting the symbol from syntax here, we know we have at least 1 operation referencing the symbol; the declaration operation. That gives us more insight to say that we can definitely give a yes/no answer, instead of "unknown" where an answer is meaningless. We could also change IsSymbolAssignedPossiblyNullValue to `bool` instead of `bool?` and return false in cases where no reference is found.